src/test/java/com/google/sps/servlets/MatchDecisionsServletTest.java

```diff
@@ -0,0 +1,306 @@
+import com.google.appengine.api.datastore.Entity;
+import org.junit.Test;
+import org.junit.After;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.junit.Before;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+
+@RunWith(JUnit4.class)
+public class MatchDecisionsServletTest {
+  private static final String MATCH_INFO_ENTITY = "match-info";
+  private static final String USER_ID_PROPERTY = "id";
+  private static final String POTENTIAL_MATCHES_PROPERTY = "potential-matches";
+  private static final String FRIENDED_IDS_PROPERTY = "friended-ids";
+  private static final String PASSED_IDS_PROPERTY = "passed-ids";
+  private static final String MATCHES_LIST_PROPERTY = "matches-list";
+
+  private static final String FRIENDED_DECISION = "FRIENDED";
+  private static final String PASSED_DECISION = "PASSED";
+
+  private static final String USER_ID_REQUEST_PARAM = "userid";
+  private static final String POTENTIAL_MATCH_REQUEST_PARAM = "potentialMatchID";
+  private static final String DECISION_REQUEST_PARAM = "decision";
+
+  private static final String TEST_USER_1_ID = "5555";
+  private static final String TEST_USER_2_ID = "1776";
+  private static final String TEST_USER_3_ID = "1234";
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Mock
+  private HttpServletRequest mockRequest;
+
+  @Mock
+  private HttpServletResponse mockResponse;
+
+  private DatastoreService datastore;
+  private MatchDecisionsServlet servletUnderTest;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+
+    helper.setUp();
+
+    servletUnderTest = new MatchDecisionsServlet();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+  
+  /**
+  * Scenario where a user chooses to friend another user as their first decision
+  *
+  * <p>Should result in the current User 1's friended list getting updated to hold just User 2.
+  */
+  @Test
+  public void firstFriendedDecision() throws IOException{
+    when(mockRequest.getParameter(USER_ID_REQUEST_PARAM)).thenReturn(TEST_USER_1_ID);
+    when(mockRequest.getParameter(POTENTIAL_MATCH_REQUEST_PARAM)).thenReturn(TEST_USER_2_ID);
+    when(mockRequest.getParameter(DECISION_REQUEST_PARAM)).thenReturn(FRIENDED_DECISION);
+
+    addTestMatchInfoToDatastore(datastore, TEST_USER_1_ID, ImmutableList.of(TEST_USER_2_ID),
+      ImmutableList.of(), ImmutableList.of(), ImmutableList.of());
+    addTestMatchInfoToDatastore(datastore, TEST_USER_2_ID, ImmutableList.of(TEST_USER_1_ID),
+      ImmutableList.of(), ImmutableList.of(), ImmutableList.of());
+
+    servletUnderTest.doPost(mockRequest, mockResponse);
```

Comment:
How do you use a mock/local datastore in the test? It looks like you're just using the actual datastore implementation.

Reply:
Daniel suggested (on my most recent PR) too make my blobstore and datastore instances in my servlet package-private, and then reassign them in the setUp method of my testing file.

Reply:
I declared datastore as a private variable, and then I'm initializing it in the setUp() method. 

I'll take a look into the package-private instances!